viewer/viewer.js

```diff
@@ -4969,138 +4973,606 @@ app.post('/removeTags', logAction(), function(req, res) {
 });
 
 //////////////////////////////////////////////////////////////////////////////////
-//// Search Session Add/Remove Tags
+//// Packet Search
 //////////////////////////////////////////////////////////////////////////////////
+function packetSearch (packet, options) {
+  if (options.searchType === 'asciicase') {
+    if (packet.toString().includes(options.search)) {
+      return true;
+    }
+  } else if (options.searchType === 'ascii') {
+    if (packet.toString().toLowerCase().includes(options.search.toLowerCase())) {
+      return true;
+    }
+  } else if (options.searchType === 'regex' && options.regex) {
+    if (packet.toString().match(options.regex)) {
+      return true;
+    }
+  } else if (options.searchType === 'hex') {
+    if (packet.toString('hex').includes(options.search)) {
+      return true;
+    }
+  } else {
+    console.error('Invalid hunt search type');
+    return false;
+  }
+
+  return false;
+}
+
+function sessionHunt (sessionId, options, cb) {
+  function packetLoop (packets) {
```

Comment:
I gave you wrong info.  This src is even/dst is odd only works for reassembled.  For non reassembled you have to check each packet which it is. :(  see reassemble_* how it does a key and checks each packet.